pypesto/result/optimize.py

```diff
@@ -164,18 +166,46 @@ def __len__(self):
 
     def append(
         self,
-        optimizer_result: OptimizerResult,
+        optimize_result: OptimizationResult,
+        sort: bool = True,
+        prefix: str = '',
     ):
         """
-        Append an optimizer result to the result object.
+        Append an OptimizerResult or an OptimizeResult to the result object.
 
         Parameters
         ----------
-        optimizer_result:
-            The result of one (local) optimizer run.
+        optimize_result:
+            The result of one or more (local) optimizer run.
+        sort:
+            Boolean used so we only sort once when appending an
+            optimize_result.
         """
-        self.list.append(optimizer_result)
-        self.sort()
+        current_ids = set(self.id)
+        if isinstance(optimize_result, OptimizeResult):
+            new_ids = [
+                prefix + identifier for identifier in optimize_result.id
+            ]
+            if current_ids.isdisjoint(new_ids):
+                raise ValueError(
+                    "Some id's you want to merge coincide with "
+                    "the existing id's. Please use an "
+                    "appropriate prefix such as 'run_2_'."
+                )
+            for optimizer_result in optimize_result.list:
+                self.append(optimizer_result, sort=False, prefix=prefix)
+        elif isinstance(optimize_result, OptimizerResult):
+            new_id = prefix + optimize_result.id
+            if new_id in current_ids:
+                raise ValueError(
+                    "The id you want to merge coincides with "
+                    "the existing id's. Please use an "
+                    "appropriate prefix such as 'run_2_'."
+                )
+            optimize_result.id = new_id
+            self.list.append(optimize_result)
```

Comment:
```suggestion
        if isinstance(optimize_result, OptimizeResult):
            for optimizer_result in optimize_result.list:
                self.append(optimizer_result, sort=False, prefix=prefix)
        elif isinstance(optimize_result, OptimizerResult):
            new_id = prefix + optimize_result.id
            if new_id in self.id:
                raise ValueError(
                    "An ID you want to merge coincides with "
                    "the existing IDs. Please set the `prefix` argument"
                    "to e.g. 'run_2_'."
                )
            optimize_result.id = new_id
            self.list.append(optimize_result)
```

Reply:
if I would do it like this what could happen is that some runs get appended cause they do have a unique id, but then one duplicate gets appended and throws an error... wouldn't it be faster/more convenient to check before?

Reply:
Ah right, since it's in-place. Perhaps appending to the same "run" should be in-place, and merging two different runs should return a new `OptimizeResult`, so that the constituent `OptimizeResult`s are left as-is? e.g. if this is used in pyPESTO behind the scenes, a user's objects should not be mutated I guess. Could still e.g. include an `inplace` argument for users that want the operation to be in-place (and agreed, if this case then a check can make sense)?